### ISSUE TITLE
Make it possible to generate interfaces instead of types

### ISSUE THREAD

**Issue by felixandersen** [dotansimha/graphql-code-generator#1810]
Using version 0.14.5 we often used interfaces generated by the tool to great effect in our TypeScript code. I'll provide a short example: 

Given the `codegen.yml`-file:
```
schema:
  - https://countries.trevorblades.com
overwrite: true
generates:
  src/app/countries-example.ts:
    documents: ./src/app/countries.graphql
    plugins:
      - typescript-common
      - typescript-client
      - typescript-apollo-angular
```

and the `src/app/countries.graphql`
```
query Countries {
  countries {
    name
  }
}
```

We can, in our Angular component, do this:
```
import { Component } from '@angular/core';

import { CountriesGQL, Countries } from './countries-example';
import { ApolloQueryResult } from 'apollo-client';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
})
export class AppComponent {
  public firstCountry: Countries.Countries;

  constructor(
    private countriesService: CountriesGQL
  ) {
    countriesService.fetch().subscribe((response: ApolloQueryResult<Countries.Query>) => {
      this.firstCountry = response.data.countries[0];
    });
  }
}
```

Now, when updating to 1.0.0 and having the following `codegen.yml`:
```
schema:
  - https://countries.trevorblades.com
overwrite: true
generates:
  src/app/countries-example.ts:
    documents: ./src/app/countries.graphql
    plugins:
      - typescript
      - typescript-operations
      - typescript-apollo-angular
```
`graphql-code-generator` generates types instead of interfaces.
This means that we have to do the following in our Angular component:

```
import { Component } from '@angular/core';

import { IconDefinition, faBell } from '@fortawesome/pro-light-svg-icons';
import { CountriesGQL, CountriesQuery } from './countries-example';
import { ApolloQueryResult } from 'apollo-client';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
})
export class AppComponent {
  public alertsAppIcon: IconDefinition = faBell;

  public firstCountry: CountriesQuery['countries'][0];

  constructor(
    private countriesService: CountriesGQL
  ) {
    countriesService.fetch().subscribe((response: ApolloQueryResult<CountriesQuery>) => {
      this.firstCountry = response.data.countries[0];
    });
  }
}
```

Which looks worse IMO

Would it maybe be possible to get the tool to generate interfaces again? Maybe a configuration variable?

Any ideas? Are we missing something obvious?

Thanks for an amazing tool!

**Comment by dotansimha** [comment]
@felixandersen The issue here is not `interface`s over `type`s. They are almost identical in TypeScript.
The issue here is that you wish to use the pre 1.0 types we generated per each query. 
We decided to drop it in order to support more environments (no `namespace`s now, so it's also possible to use it with Babel-TypeScript), and to get more maintainable code with less conflicts and mis-matches.
If you wish to get the types you had before, we created a plugin called `typescript-compatibility` that will generate the old-style namespaces and types for you. 
I recommend to use more fragments in your GraphQL queries and then it will be easier to access nested fields and types.
